core/src/test/java/kafka/server/share/SharePartitionManagerTest.java

```diff
@@ -1221,6 +1224,36 @@ public void testReplicaManagerFetchShouldProceed() {
             any(), any(), any(ReplicaQuota.class), any());
     }
 
+    @Test
+    public void testCloseSharePartitionManager() throws Exception {
+        SharePartitionManager sharePartitionManager = SharePartitionManagerBuilder.builder()
+                .withShareGroupPersister(mock(Persister.class)).build();
+
+        List<Integer> mockList = new ArrayList<>();
+        sharePartitionManager.timer().add(createTimerTask(mockList));
+        // Allowing first timer task to expire. The timer task will add an element to mockList.
+        TestUtils.waitForCondition(
+                () -> mockList.size() == 1,
+                DEFAULT_MAX_WAIT_TIMER_TASK_TIMEOUT_MS,
+                () -> "Timer task never got timed out.");
+
+        sharePartitionManager.timer().add(createTimerTask(mockList));
+        // Closing the sharePartitionManager closes timer object in sharePartitionManager.
+        sharePartitionManager.close();
+        // Allowing second timer task to expire. It won't be able to add an element to mockList.
+        Thread.sleep(DEFAULT_MAX_WAIT_TIMER_TASK_TIMEOUT_MS);
+        assertEquals(1, mockList.size());
+    }
+
+    private TimerTask createTimerTask(List<Integer> mockList) {
+        return new TimerTask(100) {
+            @Override
```

Comment:
nit: It can be a lambda method.

Reply:
since, this functionality is required at 2 places, I thought to create it as a different function `createTimerTask`

Reply:
it cannot be converted as a lambda method since Runnable task is not a parameter of TimeTask constructor.